api/grpcserver/smesher_service.go

```diff
@@ -125,31 +119,112 @@ func (s SmesherService) SetMinGas(ctx context.Context, in *pb.SetMinGasRequest)
 // PostStatus returns post data status
 func (s SmesherService) PostStatus(ctx context.Context, in *empty.Empty) (*pb.PostStatusResponse, error) {
 	log.Info("GRPC SmesherService.PostStatus")
-	return nil, status.Errorf(codes.Unimplemented, "this endpoint is not implemented")
+
+	status, err := s.post.PostStatus()
+	if err != nil {
+		return nil, err
+	}
+	return &pb.PostStatusResponse{Status: statusToPbStatus(status)}, nil
 }
 
 // PostComputeProviders returns a list of available post compute providers
 func (s SmesherService) PostComputeProviders(ctx context.Context, in *empty.Empty) (*pb.PostComputeProvidersResponse, error) {
 	log.Info("GRPC SmesherService.PostComputeProviders")
-	return nil, status.Errorf(codes.Unimplemented, "this endpoint is not implemented")
+
+	providers := s.post.PostComputeProviders()
+
+	res := &pb.PostComputeProvidersResponse{}
+	res.PostComputeProvider = make([]*pb.PostComputeProvider, len(providers))
+	for i, p := range providers {
+		hs, err := p.Benchmark()
+		if err != nil {
+			return nil, err
```

Comment:
Please don't return `err` directly here. The pattern we've been using is to pass the raw error to `log.Error` and then return `status.Error(codes.Internal, ...)` with a short description of the error fit for external consumption.